parquet-variant/src/builder.rs

```diff
@@ -195,7 +194,28 @@ impl ValueBuffer {
         self.0.len()
     }
 
-    fn append_variant<'m, 'd>(&mut self, variant: Variant<'m, 'd>) {
+    fn new_object<'a>(&'a mut self, metadata_builder: &'a mut MetadataBuilder) -> ObjectBuilder<'a> {
```

Comment:
This is the major change - move the low level appending to the `ValueBuffer` itself

Then all the recursion seems to fall into place